packages/scripts/src/remove-training-cases.js

```diff
@@ -0,0 +1,354 @@
+	await tx.caseStatus.deleteMany({
+		where: { caseId }
+	});
+	console.log(`Removing published state: ${reference}`);
+	await tx.casePublishedState.deleteMany({
+		where: { caseId }
+	});
+	console.log(`Removing grid reference: ${reference}`);
+	await tx.gridReference.deleteMany({
+		where: { caseId }
+	});
+	console.log(`Removing project team: ${reference}`);
+	await tx.projectTeam.deleteMany({
+		where: { caseId }
+	});
+	console.log(`Removing subscription: ${reference}`);
+	await tx.subscription.deleteMany({
+		where: { caseId }
+	});
+};
+
+/**
+ * @param {string} reference
+ * @returns {Promise<void>}
+ */
+const removeCase = async (reference) => {
+	try {
+		const startTime = Date.now();
+		await databaseConnector.$transaction(
+			async (tx) => {
+				const { id: caseId } = (await getCaseByRef(reference)) || {};
+				if (!caseId) {
+					throw new Error('No such case ' + reference);
+				}
+				const caseDetails = await getCaseById(caseId, {
+					applicationDetails: true,
+					applicant: true
+				});
+
+				await removeApplicationDetails(tx, caseDetails);
+				await removeExaminationTimetables(tx, caseDetails);
+				await removeRepresentations(tx, caseDetails);
+				await removeProjectUpdates(tx, caseDetails);
+				await removeS51Advices(tx, caseDetails);
+				await removeOtherAssociatedRecords(tx, caseDetails);
+				await removeFoldersAndDocuments(tx, caseDetails);
+				await removeApplicant(tx, caseDetails);
+				await removeS51Advices(tx, caseDetails);
+
+				await tx.case.delete({ where: { id: caseId } });
+
+				console.log(`${reference} Removed in ${(Date.now() - startTime) / 1000} seconds`);
+
+				if (isSimulatedTest) {
+					throw new Error('Simulated deletion test');
+				}
+			},
+			{
+				maxWait: 20000, // default: 2000
+				timeout: 100000 // default: 5000
+			}
+		);
+	} catch (e) {
+		if (e.message !== 'Simulated deletion test') {
+			console.log(reference + ' Removal failed');
+			console.log(e.message);
+			return e;
+		}
+	}
+};
+
+/**
+ * Remove cases and associated records that match the references supplied
+ *
+ * @param {string[]} references
+ * @returns {Promise<{successes,errors}>}
+ */
+export const removeCases = async (references) => {
+	const errors = [];
+	const successes = [];
+
+	if (isSimulatedTest) {
+		console.log('SIMULATED DELETION TEST, NO CASES WILL BE REMOVED\n');
+	}
+
+	for (const reference of references) {
+		const error = await removeCase(reference);
+		if (error) {
+			errors.push({ reference, error });
+		} else {
+			successes.push({ reference });
+		}
+	}
+
+	console.log('\n*************************************');
+	console.log(`Removed ${successes.length} case${successes.length === 1 ? '' : 's'}`);
+	successes.forEach(({ reference }) => console.log('- ' + reference));
+	if (errors.length > 0) {
+		console.log(`\nSkipped ${errors.length} case${errors.length === 1 ? '' : 's'} due to errors:`);
+		errors.forEach(({ reference }) => console.log('- ' + reference));
```

Comment:
Should we log the error here as well?

Reply:
The list of errors is thrown here and logged within "remove-all-training-cases-entry.js"